packages/xo-server/src/xo-mixins/file-restore-ng.mjs

```diff
@@ -108,4 +119,43 @@ export default class BackupNgFileRestore {
           adapter.listPartitionFiles(diskId, partitionId, path)
         )
   }
+
+  @decorateWith(Disposable.factory)
+  *_mountPartition(remoteId, diskId, partitionId) {
+    const adapter = yield this._app.getBackupsRemoteAdapter(remoteId)
+
+    // yield(2) the disposable to use it
+    // yield(1) the value to make it available
+    yield yield adapter.getPartition(diskId, partitionId)
+  }
+
+  async mountPartition(remoteId, diskId, partitionId) {
+    const mounts = this.#mounts
+    const key = [remoteId, diskId, partitionId]
+
+    let pDisposable = mounts.get(key)
+    if (pDisposable === undefined) {
+      pDisposable = this._mountPartition(remoteId, diskId, partitionId)
+      mounts.set(key, pDisposable)
+      pDisposable.catch(() => {
+        mounts.delete(key)
```

Comment:
Exactly, which will forward the error to the original caller.

The cache is useful to re-use already mounted partitions and to avoid race conditions.